commit 18e7a9124517abca9c8a854e7ba719ad4cf3bb03
Author: luorongfu <luorongfu@ifreegroup.com>
Date:   Tue Jan 2 13:13:04 2024 +0800

    添加到系统app中

diff --git a/app/src/main/java/com/nlscan/scantool/MainActivity.java b/app/src/main/java/com/nlscan/scantool/MainActivity.java
index 00bbbc3..507bfdf 100755
--- a/app/src/main/java/com/nlscan/scantool/MainActivity.java
+++ b/app/src/main/java/com/nlscan/scantool/MainActivity.java
@@ -46,6 +46,12 @@ import io.reactivex.android.schedulers.AndroidSchedulers;
 import io.reactivex.functions.Consumer;
 import io.reactivex.schedulers.Schedulers;
 
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.List;
+
 /**
  * Created on 2020/6/29
  *
@@ -100,10 +106,28 @@ public class MainActivity extends AppCompatActivity {
         etResult.setFocusable(false);
         etResult.setFocusableInTouchMode(false);
         setTitle("ScanTool SDK"+ds.GetSdkVersion());
+        Log.d(TAG, "ifree onCreate.");
+        try{
+               PrintWriter pw_broadcast_finish = new PrintWriter("/sys/bus/platform/devices/soc:qcom,scanner/power_gpio");
+               pw_broadcast_finish.println("1");
+               pw_broadcast_finish.close();
+               Log.d(TAG, "ifree write scanner 1");
+       }catch (Exception e) {
+               Log.e(TAG, "Error write(open) scanner.", e);
+       }
     }
 
     @Override
     protected void  onDestroy() {
+        /*
+        try{
+            PrintWriter pw_broadcast_finish = new PrintWriter("/sys/bus/platform/devices/soc:qcom,scanner/power_gpio");
+            pw_broadcast_finish.println("0");
+            pw_broadcast_finish.close();
+            Log.d(TAG, "ifree write scanner 0");
+        }catch (Exception e) {
+            Log.e(TAG, "Error write(open) scanner.", e);
+        }*/
         if (ds != null)
             ds.close();
         super.onDestroy();
@@ -290,9 +314,9 @@ public class MainActivity extends AppCompatActivity {
         Log.d(TAG, "onCommselect: "+ position);
         switch(position){
             case 0:
-                ds = new NLDevice(NLDeviceStream.DevClass.DEV_CDC);
-                editSerialname.setEnabled(false);
-                editBaud.setEnabled(false);
+                ds = new NLDevice(NLDeviceStream.DevClass.DEV_UART);
+                editSerialname.setEnabled(true);
+                editBaud.setEnabled(true);
                 break;
             case 1:
                 ds = new NLDevice(NLDeviceStream.DevClass.DEV_POS);
@@ -305,9 +329,9 @@ public class MainActivity extends AppCompatActivity {
                 editBaud.setEnabled(false);
                 break;
             case 3:
-                ds = new NLDevice(NLDeviceStream.DevClass.DEV_UART);
-                editSerialname.setEnabled(true);
-                editBaud.setEnabled(true);
+                ds = new NLDevice(NLDeviceStream.DevClass.DEV_CDC);
+                editSerialname.setEnabled(false);
+                editBaud.setEnabled(false);
                 break;
         }
     }
@@ -395,8 +419,11 @@ public class MainActivity extends AppCompatActivity {
     void OnUpdate( ) {
         int len;
 
-        if(newFwPath == null)
-            return;
+        if(newFwPath == null) {
+            Log.d(TAG, "ifree use default path");
+            newFwPath = "/system/etc/FIR.UJ101.ST.G05.1.TEST1.bin2";
+            //return;
+        }
 
         File file= new File(newFwPath);
         if (file.exists() && file.isFile()){
diff --git a/app/src/main/res/layout/activity_main.xml b/app/src/main/res/layout/activity_main.xml
index 9382cc5..19d5ad4 100755
--- a/app/src/main/res/layout/activity_main.xml
+++ b/app/src/main/res/layout/activity_main.xml
@@ -82,8 +82,8 @@
             android:layout_width="match_parent"
             android:layout_height="wrap_content"
             android:background="@android:drawable/edit_text"
-            android:text="/dev/ttyHSL1"
             android:singleLine="true"
+            android:text="/dev/ttyHSL1"
             tools:ignore="LabelFor" />
 
         <TextView
@@ -97,9 +97,9 @@
             android:layout_height="wrap_content"
             android:background="@android:drawable/edit_text"
             android:digits="0123456789"
-            android:text="9600"
             android:inputType="number"
             android:singleLine="true"
+            android:text="115200"
             tools:ignore="LabelFor" />
     </TableRow>
 
diff --git a/app/src/main/res/values/arrays.xml b/app/src/main/res/values/arrays.xml
index 53166e7..c77370a 100755
--- a/app/src/main/res/values/arrays.xml
+++ b/app/src/main/res/values/arrays.xml
@@ -6,4 +6,4 @@
         <item>USB POS</item>
         <item>USB KBW</item>
     </string-array>
-</resources>
+</resources>
\ No newline at end of file
diff --git a/nlsdk/src/main/java/com/nlscan/nlsdk/NLUartStream.java b/nlsdk/src/main/java/com/nlscan/nlsdk/NLUartStream.java
index 7a98706..73b444a 100755
--- a/nlsdk/src/main/java/com/nlscan/nlsdk/NLUartStream.java
+++ b/nlsdk/src/main/java/com/nlscan/nlsdk/NLUartStream.java
@@ -9,6 +9,7 @@ import java.io.File;
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
+import android.util.Log;
 
 /**
   * This class is implemented based on Google's official SerialPort class (https://github.com/cepr/android-serialport-api),
@@ -70,6 +71,12 @@ public class NLUartStream implements NLCommStream {
             return false;
         try {
             outputStream.write(dst, pos, length);
+	    /*
+            Log.d("ifree tiger", "dst =   " + dst.toString() + ", pos = " + pos + ", length = " + length);
+            int i;
+            for(i = 0; i < length; i++)
+                Log.d("ifree tiger", "dst" + i + " = "  + dst[i]);
+	    */
         } catch (IOException e) {
             e.printStackTrace();
         }
